core/src/main/java/com/graphhopper/util/PathSimplification.java

```diff
@@ -98,6 +98,16 @@ public PointList simplify() {
                     if (removed > 0) {
                         for (int i = 0; i < toSimplify.size(); i++) {
                             reduceLength(toSimplify.get(i), offset[i], startIntervals[i], endIntervals[i] - removed);
+                            /*
```

Comment:
Ah, I think I have now fully understood the complexity, so when there are n PathDetails we do n simplifications and roughly n²/2 updates to the PathDetail object. The only alternative would be to sort the PathDetails by first index, create a dependency graph and then being to only update the PathDetails that are necessary